### DISCUSSION TITLE
Why BlazoredModal is good?

### DISCUSSION THREAD

**Discussion by desmondinho** [Blazored/Modal#503] [Q&A]
Hello!

I've recently found out BlazoredToast and BlazoredModal, explored these repositories and tried to figure out a couple of things:

1. **Why these projects are so huge?** _Different services, components and etc._
2. **What's the benefit of initializing these components on the top level of the Application?** _App.razor / MainLayout.razor_
3. **What's the benefit of creating such big projects for such components?**
4. **Why BlazoredModal is good?** _Compared to others_

I am currently working on my own UI Components Library for Blazor and want to expand my horizons as much as possible. I hope someone or @chrissainty himself could help me find the answers.

Thanks in advance!

**Comment by chrissainty** [comment]
I honestly don't know how to answer this. I'm not sure what you mean by "big"? As in number of files? If so, the only answer I can give is, that's what it takes to build this level of UI component.

**Reply by desmondinho** [reply]
Thank you for your reply!

I just got curious because I have been using Telerik UI for Blazor in the past, and their’s modal window does not look that ‘complex’. 

I like your project. You’ve done beautiful work. I am not saying that Telerik’s one is good or better, no. But both components output the same result.

Could you please provide an answer for the 2nd and the 4th questions too?

**Reply by chrissainty** [reply]
Well, you also can't see the code behind Telerik's implementation, so I don't think you can judge how complex it is. My guess is that there will be a lot going on in the background that you won't necessarily see. But volume of code is a little irrelevant to complexity anyway. 

The component is initialised at the top of the component tree as it's designed to be global for the application. Initialising it here avoids most layout issues. As for question 4, I'm not here to do a sales pitch for you. You can see what this package offers and decide if you want to use it over another, such as Telerik.

**Reply by desmondinho** [reply]
I said that about Telerik, because I can see what's behind.

_The component is initialised at the top of the component tree as it's designed to be global for the application._
Yeah, I've found your first blog posts about them. Actually, the way you do it is cool compared to how others implement it.

_As for question 4, I'm not here to do a sales pitch for you_

My bad. I shouldn't have asked for comparison with others, but should have been more concrete. Let's just forget about that question.
Actually, what interests me the most is performance. I assume that the way you implemented a modal window functionality is more performant than just simply building modals on every page where it's needed. Since I am still quite new in Blazor, I would like to know why this way is better. Could you please shade some light on this topic please?

Thanks, @chrissainty
